src/toolchains.ts

```diff
@@ -0,0 +1,154 @@
+
+interface JdkInfo {
+  version: string;
+  vendor: string;
+  id: string;
+  jdkHome: string;
+}
+
+export async function configureToolchains(
+  version: string,
+  distributionName: string,
+  jdkHome: string,
+  toolchainId?: string
+) {
+  const vendor = core.getInput(constants.INPUT_MVN_TOOLCHAIN_VENDOR) || distributionName;
+  const id = toolchainId || `${vendor}_${version}`;
+  const settingsDirectory =
+    core.getInput(constants.INPUT_SETTINGS_PATH) || path.join(os.homedir(), constants.M2_DIR);
+  const overwriteSettings = getBooleanInput(constants.INPUT_OVERWRITE_SETTINGS, true);
+
+  await createToolchainsSettings({
+    jdkInfo: {
+      version,
+      vendor,
+      id,
+      jdkHome
+    },
+    settingsDirectory,
+    overwriteSettings
+  });
+}
+
+export async function createToolchainsSettings({
+  jdkInfo,
+  settingsDirectory,
+  overwriteSettings
+}: {
+  jdkInfo: JdkInfo;
+  settingsDirectory: string;
+  overwriteSettings: boolean;
+}) {
+  core.info(
+    `Creating ${constants.MVN_TOOLCHAINS_FILE} for JDK version ${jdkInfo.version} from ${jdkInfo.vendor}`
+  );
+  // when an alternate m2 location is specified use only that location (no .m2 directory)
+  // otherwise use the home/.m2/ path
+  await io.mkdirP(settingsDirectory);
+  const originalToolchains = await readExisting(settingsDirectory);
+  const updatedToolchains = generate(
+    originalToolchains,
+    jdkInfo.version,
+    jdkInfo.vendor,
+    jdkInfo.id,
+    jdkInfo.jdkHome
+  );
+  await write(settingsDirectory, updatedToolchains, overwriteSettings);
+}
+
+// only exported for testing purposes
+export function generate(
+  original: string,
+  version: string,
+  vendor: string,
+  id: string,
+  jdkHome: string
+) {
+  let xmlObj;
+  if (original && original.length > 0) {
+    xmlObj = xmlCreate(original)
+      .root()
+      .ele({
+        toolchain: {
+          type: 'jdk',
+          provides: {
+            version: `${version}`,
+            vendor: `${vendor}`,
+            id: `${id}`
+          },
+          configuration: {
+            jdkHome: `${jdkHome}`
+          }
+        }
+      });
+  } else
+    xmlObj = xmlCreate({
+      toolchains: {
+        '@xmlns': 'https://maven.apache.org/TOOLCHAINS/1.1.0',
+        '@xmlns:xsi': 'https://www.w3.org/2001/XMLSchema-instance',
+        '@xsi:schemaLocation':
+          'https://maven.apache.org/TOOLCHAINS/1.1.0 https://maven.apache.org/xsd/toolchains-1.1.0.xsd',
+        toolchain: [
+          {
+            type: 'jdk',
+            provides: {
+              version: `${version}`,
+              vendor: `${vendor}`,
+              id: `${id}`
+            },
+            configuration: {
+              jdkHome: `${jdkHome}`
+            }
+          }
+        ]
+      }
+    });
+
+  return xmlObj.end({
+    format: 'xml',
+    wellFormed: false,
+    headless: false,
+    prettyPrint: true,
+    width: 80
+  });
+}
+
+async function readExisting(directory: string) {
```

Comment:
I think the function name should be clearer

Reply:
Changed to `readExistingToolchainsFile`